queue/driver.go

```diff
@@ -18,10 +18,9 @@ type remoteQueueDriver interface {
 	// Get finds a job by job ID. For retryable jobs, this returns the latest
 	// job attempt.
 	Get(context.Context, string) (amboy.Job, error)
-	// GetAttempt returns a job by job ID and attempt number. Only applicable to
-	// retryable jobs. If used for a non-retryable job, this should return nil
-	// and an error.
-	GetAttempt(ctx context.Context, id string, attempt int) (amboy.RetryableJob, error)
+	// GetAttempt returns a retryable job by job ID and attempt number. If used
+	// to find a non-retryable job, this should return nil job and an error.
```

Comment:
Since we are now merging the `RetryableJob` interface with the `Job` interface, would it make more sense for this just to return `nil` and no error when the job is not retryable?

Reply:
I don't think it should because squashing legitimate programatic errors (i.e. violating the doc and trying to get a non-retryable job) seems like the wrong move (I also don't like how Evergreen inconsistently returns nil, nil rather than just propagating the error from MongoDB). I also don't want to encourage incorrect usage of the retryable functionality.